### PR TITLE
Simplify the passing of sandbox options around and reduce modality.

### PR DESCRIPTION
**Pull request by calebbrown**
Split the sandbox option factory function into two - the mode was always a static arg and never a dynamic variable, so it isn't necessary. This allows for a reduction in state.

Additionally the logic to build up the option-specific args is now done prior to merging with the sandbox specific args.

Also updates `go.work.sum`

### CODE REVIEW

File: cmd/analyze/main.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func staticAnalysis(pkg *pkgmanager.Pkg) {
 		sandbox.InitNetwork()
 	}
 
-	sbOpts := makeSandboxOptions(analysis.Static)
+	sbOpts := append(worker.DynamicSandboxOptions(), makeSandboxOptions()...)
```
Comment (maxfisher-g) [inline cmd/analyze/main.go:124]: Should be `worker.StaticSandboxOptions()` here

File: go.work.sum [hunk 2]
```diff
@@ -0,0 +1,6 @@
+cloud.google.com/go/apigeeregistry v0.4.0/go.mod h1:EUG4PGcsZvxOXAdyEghIdXwAEi/4MEaoqLMLDMIwKXY=
```
Comment (maxfisher-g) [inline go.work.sum:1]: Is this file change related to the sandbox logic?

**Review by maxfisher-g — CHANGES_REQUESTED** [review]

**Comment by maxfisher-g** [comment]
Since we're modifying this, is there any way we can pull all of the sandbox logic out of the main.go files and into `internal/worker/sandbox_options.go`? And just pass in the image tag, nopull and mount info.

File: cmd/analyze/main.go (hunk 1 above)
Comment (calebbrown) [reply cmd/analyze/main.go]: Fixed

File: go.work.sum (hunk 2 above)
Comment (calebbrown) [reply go.work.sum:1]: No, it's related to the introduction of the `go.work` file. I was surprised it wasn't added when the workspace was added.

Comment (maxfisher-g) [reply go.work.sum:1]: Ah I see. Can we add a comment about this to the PR description?

**Comment by calebbrown** [comment]
> Since we're modifying this, is there any way we can pull all of the sandbox logic out of the main.go files and into two functions in `internal/worker/sandbox_options.go`? And just pass in the image tag, nopull and mount info.

I don't think that's a great idea, as the whole point of sandbox.Option is to avoid function calls like `generateOptions(imageTag, noPull, mounts...)`.

Also long-term I'm thinking we should refactor the dynamic analysis and static analysis logic into a `LoggingSandbox` and a regular non-logging `Sandbox`, with a generalised option for enabling/disabling network isolation to reduce the need for option factories.

The idea is that a `LoggingSandbox` attaches artifacts like pcaps, strace, file data, http(s) request, stdio capture, etc to the result, where `Sandbox` doesn't, but can be used to run potentially dangerous workloads (e.g. static analysis, composer.phar package downloading).

An alternative would be to have "official" option factories called `sandbox.LoggingOptions()` or `sandbox.DefaultOptions()`

File: go.work.sum (hunk 2 above)
Comment (calebbrown) [reply go.work.sum:1]: I added a very brief comment

**Review by oliverchang — APPROVED** [review]

File: go.work.sum (hunk 2 above)
Comment (maxfisher-g) [reply go.work.sum:1]: Thanks

**Review by maxfisher-g — APPROVED** [review]
